python/lsst/ts/watcher/rules/cp_verify_alarm.py

```diff
@@ -0,0 +1,266 @@
+        )
+        super().__init__(
+            config=config,
+            name=f"CpVerifyAlarm.{remote_info.name}:{remote_info.index}",
+            remote_info_list=[remote_info],
+        )
+
+        self.calibration_type = config.calibration_type
+
+    @classmethod
+    def get_schema(cls):
+        schema_yaml = """
+            $schema: http://json-schema.org/draft-07/schema#
+            description: Configuration for CpVerifyBiasAlarm
+            type: object
+            properties:
+                calibration_type:
+                    description: >-
+                        Alarm calibration type: BIAS, DARK, or FLAT.
+                    type: string
+                ocps_index:
+                    OCPS index (e.g., OCPS:1: LATISS; OCPS:2: LSSTComCam).
+                verification_threshold:
+                    type: integer
+                    descriptor: Maximum number of failures per detector per test type.
+                default: 8
+            required: [name]
+            additionalProperties: false
+        """
+        return yaml.safe_load(schema_yaml)
+
+    def __call__(self, topic_callback):
+        msg = topic_callback.get()
+        # OCPS response after calling the pipetask
+        response = json.loads(msg.result)
+        # boolean: did verification fail?
+        verify_tests_pass = await self.get_verify_tests_pass_boolean(response)
+
+        if verify_tests_pass:
+            return watcher.NoneNoReason
+        else:
+            # Tests did not pass. Follow up on images and calibrations is
+            # recommended.
+            return AlarmSeverity.SERIOUS, "FAULT state"
+
+    async def get_verify_tests_pass_boolean(self, response_verify):
+        """Determine if cp_verify bias tests passed from OCPS response.
+
+        Parameters
+        ----------
+        response : `dict`
+            OCPS call response.
+
+        Returns
+        -------
+        verify_pass : `bool`
+            Did the cp_verify tests pass?
+        """
+        verify_pass = True
+        # Loop over the entries of the 'results' list
+        # and look for the adecuate dataset type.
+        for entry in response_verify["results"]:
+            if "verifyBiasStats" in entry["uri"]:
+                verify_stats_string = "verifyBiasStats"
+                break
+            elif "verifyDarkStats" in entry["uri"]:
+                verify_stats_string = "verifyDarkStats"
+                break
+            elif "verifyFlatStats" in entry["uri"]:
+                verify_stats_string = "verifyFlatStats"
+                break
+            else:
+                # This is not a response from cp_verify
+                # bias, dark, or flat.
+                # Do not issue a watcher alarm.
+                return verify_pass
+
+        # Find repo and instrument
+        for entry in response_verify["parameters"]["environment"]:
+            if entry["name"] == "BUTLER_REPO":
+                repo = entry["value"]
```

Comment:
`repo` is not defined outside this `if` statement, so if this never runs you will end up with undefined `repo`... 

You probably want to check for this condition.

Reply:
We know that the OCPS response contains this entry, but if not, I will raise an ``RuntimeError`` (we can't do anything if the response, for whatever reason, does not have a repo).

Reply:
what if `response_verify["parameters"]["environment"]` is empty?

Reply:
I added a check for its length, and issue an error if is empty.